corehq/apps/hqwebapp/static/hqwebapp/js/components/select_toggle.js

```diff
@@ -0,0 +1,54 @@
+/*
+ * Component for displaying a selection as a set of buttons.
+ * Creates an invisible <select> element to store the value and a set of buttons for the user to interact with.
+ *
+ * Required parameters
+ *  - options: Data to build select options. May be list of strings or list of objects with `id` and `text` properties.
+ *
+ * Optional parameters
+ *  - name: Name of select
+ *  - id: ID of select
+ *  - value: initial value of select
+ */
+
+hqDefine('hqwebapp/js/components/select_toggle', [
+    'jquery',
+    'knockout',
+    'underscore',
+], function (
+    $,
+    ko,
+    _
+) {
+    return {
+        viewModel: function (params) {
+            var self = this;
+
+            // Attributes passed on to the input
+            self.name = params.name || '';
+            self.id = params.id || '';
+
+            // Data
+            self.value = ko.observable(params.value || '');
+            self.options = ko.observableArray(_.map(params.options, function (o) {
```

Comment:
👀 single letter variable

Reply:
eh, I don't mind them in one-statement callbacks like this

Reply:
You call _that_ (function spanning lines 33-50) a one-statement callback? Granted, this variable is only used on the first three lines, but I had to scan the whole function to figure that out.

Reply:
I responded to this comment based on the email, which only showed up to this line. Forgot how long this callback is. I don't love how big this function got, or the fact that it contains an anonymous knockout model, but I'm going to leave it alone for now.